src/gen_zp_template.cc

```diff
@@ -154,6 +163,17 @@ int main(int argc, char *argv[])
         return 0;
     }
 
+    fgets(line, 160, config_file);
```

Comment:
Stick bool here to check if resolution histos should be produced. The `fgets` returns line for success and null ptr for failure. Check for null ptr and empty string (since sometimes the config has extra empty lines) to set the flag. Wrap the next two changes in the `if` that checks this flag.